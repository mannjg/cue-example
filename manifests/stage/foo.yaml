apiVersion: apps/v1
kind: Deployment
metadata:
  name: foo
  namespace: staging
  labels:
    app: foo
    component: backend
    managed: cue
spec:
  replicas: 2
  selector:
    matchLabels:
      app: foo
      component: backend
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 0
  template:
    metadata:
      labels:
        app: foo
        component: backend
        managed: cue
      annotations:
        prometheus.io/scrape: "true"
        prometheus.io/port: "8080"
        prometheus.io/path: /metrics
    spec:
      containers:
        - name: foo
          image: foo:stage-v1.2.3-rc1
          env:
            - name: APP_ENV
              value: production
            - name: APP_PORT
              value: "8080"
            - name: DATABASE_HOST
              value: postgres.database.svc.cluster.local
            - name: DATABASE_PORT
              value: "5432"
            - name: DATABASE_NAME
              value: foo
            - name: DATABASE_USER
              valueFrom:
                secretKeyRef:
                  name: foo-secrets
                  key: db-user
            - name: DATABASE_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: foo-secrets
                  key: db-password
            - name: REDIS_URL
              valueFrom:
                configMapKeyRef:
                  name: foo-config
                  key: redis-url
            - name: LOG_LEVEL
              valueFrom:
                configMapKeyRef:
                  name: foo-config
                  key: log-level
          envFrom:
            - configMapRef:
                name: foo-config
            - secretRef:
                name: foo-secrets
          imagePullPolicy: Always
          livenessProbe:
            httpGet:
              path: /health/live
              port: 8080
              scheme: HTTP
            initialDelaySeconds: 30
            periodSeconds: 10
            timeoutSeconds: 5
            successThreshold: 1
            failureThreshold: 3
          ports:
            - name: http
              containerPort: 8080
              protocol: TCP
          readinessProbe:
            httpGet:
              path: /health/ready
              port: 8080
              scheme: HTTP
            initialDelaySeconds: 10
            periodSeconds: 5
            timeoutSeconds: 3
            successThreshold: 1
            failureThreshold: 3
          resources:
            requests:
              cpu: 250m
              memory: 512Mi
            limits:
              cpu: 500m
              memory: 1Gi
          volumeMounts:
            - name: data
              mountPath: /var/lib/myapp/data
              readOnly: false
            - name: config
              mountPath: /etc/myapp/config
              readOnly: true
            - name: cache
              mountPath: /var/cache/myapp
              readOnly: false
          securityContext:
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
            readOnlyRootFilesystem: false
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - ALL
      volumes:
        - name: data
          persistentVolumeClaim:
            claimName: foo-data
        - name: config
          configMap:
            name: foo-config
        - name: cache
          emptyDir:
            medium: Memory
            sizeLimit: 256Mi
      securityContext:
        runAsGroup: 1000
        runAsNonRoot: true
        runAsUser: 1000
        fsGroup: 1000
      serviceAccountName: foo
---
apiVersion: v1
kind: Service
metadata:
  name: foo
  namespace: staging
  labels:
    app: foo
    component: backend
    managed: cue
spec:
  type: ClusterIP
  selector:
    app: foo
    component: backend
  ports:
    - name: http
      protocol: TCP
      port: 80
      targetPort: 8080
  sessionAffinity: None
