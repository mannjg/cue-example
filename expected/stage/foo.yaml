apiVersion: apps/v1
kind: Deployment
metadata:
  name: foo
  namespace: staging
  labels:
    app: foo
    deployment: foo
spec:
  replicas: 2
  selector:
    matchLabels:
      app: foo
      deployment: foo
  template:
    metadata:
      labels:
        app: foo
        deployment: foo
    spec:
      containers:
        - name: foo
          image: foo:stage-v1.2.3-rc1
          envFrom: []
          imagePullPolicy: Always
          livenessProbe:
            httpGet:
              path: /health/live
              port: 8080
              scheme: HTTP
            initialDelaySeconds: 30
            periodSeconds: 10
            timeoutSeconds: 5
            failureThreshold: 3
          ports:
            - name: http
              containerPort: 8080
              protocol: TCP
          readinessProbe:
            httpGet:
              path: /health/ready
              port: 8080
              scheme: HTTP
            initialDelaySeconds: 10
            periodSeconds: 5
            timeoutSeconds: 3
            failureThreshold: 3
          resources:
            requests:
              cpu: 250m
              memory: 512Mi
            limits:
              cpu: 500m
              memory: 1Gi
          volumeMounts:
            - name: data
              mountPath: /var/lib/myapp/data
              readOnly: false
            - name: config
              mountPath: /etc/myapp/config
              readOnly: true
            - name: cache
              mountPath: /var/cache/myapp
              readOnly: false
            - name: projected-secrets
              mountPath: /var/secrets
              readOnly: true
          securityContext:
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
            readOnlyRootFilesystem: false
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - ALL
      volumes:
        - name: data
          persistentVolumeClaim:
            claimName: foo-data
        - name: config
          configMap:
            name: foo-config
        - name: cache
          emptyDir:
            medium: Memory
            sizeLimit: 256Mi
        - name: projected-secrets
          projected:
            defaultMode: 256
            sources:
              - secret:
                  name: foo-stage-secrets
                  items:
                    - key: db-user
                      path: database/username
                    - key: db-password
                      path: database/password
              - configMap:
                  name: foo-stage-config
                  items:
                    - key: redis-url
                      path: config/redis-url
              - configMap:
                  name: stage-cluster-ca
                  items:
                    - key: ca.crt
                      path: config/cluster-ca.crt
              - downwardAPI:
                  items:
                    - path: pod/name
                      fieldRef:
                        fieldPath: metadata.name
                    - path: pod/namespace
                      fieldRef:
                        fieldPath: metadata.namespace
      securityContext:
        runAsGroup: 1000
        runAsNonRoot: true
        runAsUser: 1000
        fsGroup: 1000
      serviceAccountName: foo
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
---
apiVersion: v1
kind: Service
metadata:
  name: foo
  namespace: staging
  labels:
    app: foo
    deployment: foo
spec:
  type: ClusterIP
  selector:
    app: foo
    component: backend
  ports:
    - name: http
      protocol: TCP
      port: 80
      targetPort: 8080
  sessionAffinity: None
